src/main/java/com/example/gruppajava/controller/ParkingEventController.java

```diff
@@ -0,0 +1,99 @@
+package com.example.gruppajava.controller;
+
+import java.net.URI;
+import java.time.LocalDateTime;
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import com.example.gruppajava.entity.ParkingEvent;
+import com.example.gruppajava.repository.CarRepository;
+import com.example.gruppajava.repository.ParkingEventRepository;
+import com.example.gruppajava.repository.UserRepository;
+
+
+@RestController
+@RequestMapping("/api/")
+public class ParkingEventController {
+    @Autowired ParkingEventRepository parkingEventRepository;
+    @Autowired CarRepository carRepository;
+    @Autowired UserRepository userRepository;
+
+    // Get all parking events.
+    @GetMapping("parkingevents")
+    public Iterable<ParkingEvent> getAllParkingEvents() {
+        return parkingEventRepository.findAll();
+    }
+
+    // Create a new parking event.
+    @PostMapping("parkingevents")
+    public ResponseEntity<ParkingEvent> addParkingEvent(@RequestBody Map<String, String> body) {
+        Long userId = Long.parseLong(body.get("userId"));
+        Long carId = Long.parseLong(body.get("carId"));
+        int parkingslotId = Integer.parseInt(body.get("parkingslotId"));
+
+        ParkingEvent parkingEvent = new ParkingEvent();
+        parkingEvent.setUser(userRepository.findById(userId).get());
+        parkingEvent.setCar(carRepository.findById(carId).get());
+        parkingEvent.setSlotId(parkingslotId);
+        parkingEvent.setStartTime(LocalDateTime.now());
+        parkingEvent.setEndTime(LocalDateTime.now().plusMinutes(10));
+        parkingEvent.setActive(true);
+        parkingEvent.setPaid(false);
+        parkingEventRepository.save(parkingEvent);
+        
+        URI location = ServletUriComponentsBuilder
+        .fromCurrentRequest()
+        .path("/{id}")
+        .buildAndExpand(parkingEvent.getId())
+        .toUri();
+
```

Comment:
Nice! Let's ask why :D

Reply:
Ask who?

Reply:
Martin, I think "response entity" just a best practice thing, me and Marcus has been getting heated for weeks about this :P

Reply:
"As a good programming practice, it’s always required that server response should contain accurate headers and status code. Without ResponseEntity, you can control the response body but it will be hard to control the headers and status code."

So, yeah, best practice stuff mostly

Reply:
okay cool!